flask_appbuilder/models/sqla/interface.py

```diff
@@ -516,9 +516,11 @@ def get(self, id, filters=None):
         return self.session.query(self.obj).get(id)
 
     def get_pk_name(self):
-        for col_name in self.list_columns.keys():
-            if self.is_pk(col_name):
-                return col_name
+        pks = [pk.name for pk in self.obj.__mapper__.primary_key]
```

Comment:
The name `get_pk_name` is becoming a bit misleading as it may contain a name or a list. I'm not big on methods that return heterogenous types.

I'd go with renaming it `get_pk` and make sure it's always a list and handled as a list by the callers.

Reply:
👍  cool, i originally kept it as str to minimize breakage (by keeping backward compatibility), but maybe a more consistent return type is better